packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -410,7 +410,7 @@ export class AuthTransformer extends TransformerAuthBase implements TransformerA
       if (context.metadata.has(indexKeyName)) {
         context.metadata.get<Set<string>>(indexKeyName)!.forEach((index) => {
           const [indexName, indexQueryName] = index.split(':');
-          this.protectListResolver(context, def, def.name.value, indexQueryName, acm, indexName);
+          this.protectListResolver(context, def, 'Query', indexQueryName, acm, indexName);
```

Comment:
What's the context on this change?

Reply:
Or rather, I understand teh change, but don't understand how things work today if that's not changed.

Reply:
Today the resolvers that are generated from various transformers are managed by the [TransformerResolversManagerProvider](https://github.com/aws-amplify/amplify-category-api/blob/25b65238b5e35e03b1275f432693d8dd6c8fc966/packages/amplify-graphql-transformer-interfaces/src/transformer-context/transformer-resolver-provider.ts#L25) where you can see that we treat `typeName.fieldName` as the key to [add/get any resolver](https://github.com/aws-amplify/amplify-category-api/blob/25b65238b5e35e03b1275f432693d8dd6c8fc966/packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts#L92). However, if we look at the [resolver interface itself](https://github.com/aws-amplify/amplify-category-api/blob/25b65238b5e35e03b1275f432693d8dd6c8fc966/packages/amplify-graphql-transformer-core/src/transformer-context/resolver.ts#L136), this is where we actually store which typeName, fieldName it corresponds to. For [most other transformers](https://github.com/aws-amplify/amplify-category-api/blob/25b65238b5e35e03b1275f432693d8dd6c8fc966/packages/amplify-graphql-model-transformer/src/resources/model-resource-generator.ts#L136), the typeName, fieldName used in the key matches the ones in the resolver itself, except for index where I found the mis-match. In this PR, we're changing that key for index query resolver which is basically used to get the resolver and append to it's slots etc from the Auth transformer and refersTo transformer.